osidb_bindings/templates_0.22.0/endpoint_module.py.jinja

```diff
@@ -0,0 +1,216 @@
+{#
+    This is a custom template derived from:
+    https://github.com/openapi-generators/openapi-python-client/blob/v0.22.0/openapi_python_client/templates/endpoint_module.py.jinja
+#}
+from http import HTTPStatus
+from typing import Any, Optional, Union, cast
+
+{# CHANGE START (1) - switch out httpx for requests #}
+{# import httpx #}
+import requests
+{# CHANGE END (1) #}
+
+from ...client import AuthenticatedClient, Client
+{# CHANGE START (2) - add missing Unset import #}
+from ...types import Response, UNSET, Unset
+{# CHANGE END (2) #}
+from ... import errors
+
+{% for relative in endpoint.relative_imports | sort %}
+{{ relative }}
+{% endfor %}
+
+{% from "endpoint_macros.py.jinja" import header_params, cookie_params, query_params,
+    arguments, client, kwargs, parse_response, docstring, body_to_kwarg %}
+
+{% set return_string = endpoint.response_type() %}
+{% set parsed_responses = (endpoint.responses | length > 0) and return_string != "Any" %}
+
+{# CHANGE START (3) - expose endpoint query parameters #}
+QUERY_PARAMS = {
+    {% for parameter in endpoint.query_parameters %}
+    "{{ parameter.name }}": {{ parameter.get_type_string(no_optional=True) }},
+    {% endfor %}
+}
+{# CHANGE END (3) #}
+
+{# CHANGE START (4) - expose request body type if exists #}
+{% if endpoint.bodies | length >= 1 %}
+REQUEST_BODY_TYPE = {{ endpoint.bodies[0].prop.get_type_string() }}
+{% endif %}
+{# CHANGE END (4) #}
+
+def _get_kwargs(
+{# CHANGE START (5) - include client in arguments #}
+    {{ arguments(endpoint, include_client=True) | indent(4) }}
+{# CHANGE END (5) #}
+) -> dict[str, Any]:
+    {{ header_params(endpoint) | indent(4) }}
+
+    {{ cookie_params(endpoint) | indent(4) }}
+
+    {{ query_params(endpoint) | indent(4) }}
+
+    _kwargs: dict[str, Any] = {
+        {# CHANGE START (6) - don't include method in kwargs #}
+        {# "method": "{{ endpoint.method }}", #}
+        {# CHANGE END (6) #}
+        {% if endpoint.path_parameters %}
+        {# CHANGE START (7) - use base URL from client #}
+        "url": f"{client.base_url}/{{ endpoint.path }}".format(
+        {# CHANGE END (7) #}
+        {%- for parameter in endpoint.path_parameters -%}
+        {{parameter.python_name}}={{parameter.python_name}},
+        {%- endfor -%}
+        ),
+        {% else %}
+        {# CHANGE START (8) - use base URL from client #}
+        "url": f"{client.base_url}{{ endpoint.path }}",
```

Comment:
missing slash?
```suggestion
        "url": f"{client.base_url}/{{ endpoint.path }}",
```

Reply:
Also I wonder what happens if the `base_url` already has a trailing slash

Reply:
Good catch. I am trimming the trailing slashes in the `session.py` so it is covered.